product-service/handlers/getProductById.ts

```diff
@@ -1,26 +1,35 @@
-import fetch from "node-fetch";
+import {Pool} from "pg";
 import {APIGatewayProxyHandler} from "aws-lambda";
-import products from "../products";
-import headers from "./headers";
+import {headers, internalError} from "./helpers";
+import {DB_OPTIONS} from "./db";
+
+// cached variable
+let pool
 
 export const getProductById: APIGatewayProxyHandler = async (event) => {
+    console.log(event, event.pathParameters)
+    // Check if pool already exists. Create new one if not.
+    if (!pool) {
+        pool = new Pool(DB_OPTIONS);
+    }
+
+    // Get client from pool
+    const client = await pool.connect();
```

Comment:
Why did you use pool? As I understood from previous lecture it's not recommended way, because you could easily overcome number of connections in pool

Reply:
@SkorykV I don't have any experience with databases. So I don't know exactly the difference between them. I just found this [q&a](https://qna.habr.com/q/525098) and I thought it will be better to use. Also I wanted to reuse DB connection logic but didn't find the way to do it